src/tools/tidy/src/edition.rs

```diff
@@ -0,0 +1,58 @@
+//! Tidy check to ensure that crate `edition` is '2018'
+//!
+
+use std::fs;
+use std::path::Path;
+
+fn filter_dirs(path: &Path) -> bool {
+    if super::filter_dirs(path) {
+        return true;
+    }
+    // FIXME: remove this whitelist
+    let skip = [
+        "src/doc/book/second-edition",
+        "src/doc/book/2018-edition",
+        "src/doc/book/ci/stable-check",
+        "src/doc/reference/stable-check",
+        "src/test",
+    ];
+    skip.iter().any(|p| path.ends_with(p))
+}
+
+pub fn check(path: &Path, bad: &mut bool) {
+    if filter_dirs(path) {
+        return;
+    }
+    for entry in t!(path.read_dir(), path).map(|e| t!(e)) {
```

Comment:
If you look in other files, we should be able to reuse the central read_dir infrastructure rather than rolling our own.

Reply:
Ok, rewritten.